service/sliceqosConfig_webhook_validation.go

```diff
@@ -0,0 +1,47 @@
+package service
+
+import (
+	"context"
+
+	controllerv1alpha1 "github.com/kubeslice/kubeslice-controller/apis/controller/v1alpha1"
+	workerv1alpha1 "github.com/kubeslice/kubeslice-controller/apis/worker/v1alpha1"
+	"github.com/kubeslice/kubeslice-controller/util"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// ValidateProjectCreate is a function to validate the creation of project
+func ValidateSliceqosConfigDelete(ctx context.Context, sliceQoSConfig *controllerv1alpha1.SliceQoSConfig) error {
+	var allErrs field.ErrorList
+	exists, err := validateIfQosProfileExists(ctx, sliceQoSConfig)
+	if err != nil {
+		return err
+	}
+	if exists {
+		err := field.Forbidden(field.NewPath("Sliceqos"), "The SliceqosProfile "+sliceQoSConfig.Name+" cannot be deleted. It is present on workerslices")
+		allErrs = append(allErrs, err)
+	}
+	//	allErrs = append(allErrs, err)
+	if len(allErrs) == 0 {
+		return nil
+	}
+	return apierrors.NewInvalid(schema.GroupKind{Group: "controller.kubeslice.io", Kind: "SliceQosConfig"}, sliceQoSConfig.Name, allErrs)
+}
+
+/* validateIfQosProfileExists function to check if qos profile exists on any of workerslices */
+func validateIfQosProfileExists(ctx context.Context, sliceQosConfig *controllerv1alpha1.SliceQoSConfig) (bool, error) {
```

Comment:
Name of the function should be something like `validateIfWorkerSliceConfigsExist`, right? We are checking WorkerSliceConfig, not the QoSProfile here.

Reply:
This function is checking the list of workerSliceConfig only. So, for me `validateIfWorkerSliceConfigsExist` seems more clear.